proofs/Settlement.thy

```diff
@@ -0,0 +1,160 @@
+theory Settlement
+
+imports Main
+
+begin
+
+text "This file contains an analysis of the settlement algorithm in the TokenNetwork contract.
+https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-and-settlement-algorithm-analysis
+The result so far confirms two things:
+  lemma s1_correct: The value calculated as 'S1 = RmaxP1 - SL2' is equal to 'S1 = D1 - W1 + T2 - T1 - L1'.
+  lemma s2_correct: Similarly for 'S2 = RmaxP2 - SL1' and 'S2 = D2 - W2 + T1 - T2 - L2'.
+The required conditions appear in the statements of the lemma.
```

Comment:
With the notes that I already made in person:
- `s1_correct` & `s2_correct` - stand only if both balance proofs are `valid` & `last`; indeed, it should be always correct in this case.
- for cases where **at least** one balance proof is `old`, we need to move the check down the channel lifecycle: after both `unlock`s are done. In these cases, after settlement you can frequently have a different `S1` or `S2` value in the contract than `S1 = D1 - W1 + T2 - T1 - L1`, due to under/overflows. We need to make sure this difference balances out with the `unlock`s. 
  - Python tests reference: https://github.com/raiden-network/raiden-contracts/blob/76758d271059c96a8781754df3c599c4c948a2de/raiden_contracts/tests/test_channel_settle_unlock_state.py#L232-L244

Reply:
I have an issue about extending the Isabelle proof for older balance proofs https://github.com/raiden-network/spec/issues/207 .